include/boost/geometry/algorithms/detail/buffer/buffered_piece_collection.hpp

```diff
@@ -191,8 +191,8 @@ struct buffered_piece_collection
 
         // 1: half, part of offsetted_rings
         segment_identifier first_seg_id;
-        int last_segment_index; // no segment-identifier - it is the same as first_seg_id
-        int offsetted_count; // part in robust_ring which is part of offsetted ring
+        signed_index_type last_segment_index; // no segment-identifier - it is the same as first_seg_id
+        signed_index_type offsetted_count; // part in robust_ring which is part of offsetted ring
```

Comment:
Well, it's initialized with `-1` (Is that correct?). So I'm guessing that it should be a signed type (unless it shouldn't be initialized to `-1`). And it's again, iterated in a loop iterating through `pc.robust_turns` elements so it should be at least of size of the `std::vector<robust_turn>::size_type`.